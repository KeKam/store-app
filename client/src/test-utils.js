import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';

import rootReducer from './redux/root-reducer';

// const render = (
//   component,
//   {
//     initialState,
//     store = createStore(rootReducer, initialState),
//     ...renderOptions
//   } = {}
// ) => {
//   return {
//     ...rtlRender(<Provider store={store}>{component}</Provider>),
//     store,
//   };
//   // const Wrapper = ({ children }) => {
//   //   return <Provider store={store}>{children}</Provider>;
//   // };
//   // return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
// };

const render = (component, { initialState } = {}) => {
  const actions = [];
  const observerMiddleware = () => (next) => (action) => {
    actions.push(action);
    return next(action);
  };
  const store = createStore(
    rootReducer,
    initialState,
    applyMiddleware(observerMiddleware)
  );
  const utils = {
    dispatch(action) {
      return store.dispatch(action);
    },
    getDispatchedActions() {
      return actions;
    },
    getState() {
      return store.getState();
    },
  };
  return {
    ...rtlRender(<Provider store={store}>{component}</Provider>),
    ...utils,
  };
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { render };
