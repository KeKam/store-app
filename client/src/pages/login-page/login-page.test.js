import React from 'react';
import '@testing-library/jest-dom';

import { render, screen } from '../../test-utils';
import LoginPage from './login-page';

describe('LoginPage component', () => {
  it('should render LoginPage component', () => {
    const { asFragment } = render(<LoginPage />);

    expect(asFragment()).toMatchSnapshot();
    expect(
      screen.getByRole('heading', { name: 'I already have an account' })
    ).toBeInTheDocument();
    expect(
      screen.getByText('Sign in with your email and password')
    ).toBeInTheDocument();
    expect(screen.getByTestId('email-input')).toHaveDisplayValue('');
    expect(screen.getByTestId('password-input')).toHaveDisplayValue('');
    expect(screen.getByRole('button', { name: 'Sign in' })).toBeInTheDocument();
    expect(
      screen.getByRole('button', { name: 'Sign in with Google' })
    ).toBeInTheDocument();
    expect(
      screen.getByRole('heading', { name: 'I do not have a account' })
    ).toBeInTheDocument();
    expect(
      screen.getByText('Sign up with your email and password')
    ).toBeInTheDocument();
    expect(screen.getByTestId('display-name')).toHaveDisplayValue('');
    expect(screen.getByTestId('email')).toHaveDisplayValue('');
    expect(screen.getByTestId('password')).toHaveDisplayValue('');
    expect(screen.getByTestId('confirm-password')).toHaveDisplayValue('');
    expect(screen.getByRole('button', { name: 'SIGN UP' })).toBeInTheDocument();
  });
});
