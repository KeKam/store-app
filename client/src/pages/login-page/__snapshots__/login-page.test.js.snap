// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage component should render LoginPage component 1`] = `
<DocumentFragment>
  <div
    class="sc-dnqmqq ckufaP"
  >
    <div
      class="sc-ifAKCX llvkyD"
    >
      <h2
        class="sc-EHOje cciXSq"
      >
        I already have an account
      </h2>
      <span>
        Sign in with your email and password
      </span>
      <form>
        <div
          class="sc-bdVaJa eEzrUY"
        >
          <input
            class="sc-bwzfXH eKtSpl"
            data-testid="email-input"
            name="email"
            required=""
            type="email"
            value=""
          />
          <label
            class="sc-htpNat gLmdTI"
            value=""
          >
            Email
          </label>
        </div>
        <div
          class="sc-bdVaJa eEzrUY"
        >
          <input
            class="sc-bwzfXH eKtSpl"
            data-testid="password-input"
            name="password"
            required=""
            type="password"
            value=""
          />
          <label
            class="sc-htpNat gLmdTI"
            value=""
          >
            Password
          </label>
        </div>
        <div
          class="sc-bZQynM iXkwVI"
        >
          <button
            class="sc-bxivhb jzXiqt"
            type="submit"
          >
             Sign in 
          </button>
          <button
            class="sc-bxivhb gxnkwT"
            type="button"
          >
            Sign in with Google
          </button>
        </div>
      </form>
    </div>
    <div
      class="sc-gzVnrw dykNtz"
    >
      <h2
        class="sc-htoDjs fyUvWz"
      >
        I do not have a account
      </h2>
      <span>
        Sign up with your email and password
      </span>
      <form>
        <div
          class="sc-bdVaJa eEzrUY"
        >
          <input
            class="sc-bwzfXH eKtSpl"
            data-testid="display-name"
            name="displayName"
            required=""
            type="text"
            value=""
          />
          <label
            class="sc-htpNat gLmdTI"
            value=""
          >
            Display Name
          </label>
        </div>
        <div
          class="sc-bdVaJa eEzrUY"
        >
          <input
            class="sc-bwzfXH eKtSpl"
            data-testid="email"
            name="email"
            required=""
            type="email"
            value=""
          />
          <label
            class="sc-htpNat gLmdTI"
            value=""
          >
            Email
          </label>
        </div>
        <div
          class="sc-bdVaJa eEzrUY"
        >
          <input
            class="sc-bwzfXH eKtSpl"
            data-testid="password"
            name="password"
            required=""
            type="password"
            value=""
          />
          <label
            class="sc-htpNat gLmdTI"
            value=""
          >
            Password
          </label>
        </div>
        <div
          class="sc-bdVaJa eEzrUY"
        >
          <input
            class="sc-bwzfXH eKtSpl"
            data-testid="confirm-password"
            name="confirmPassword"
            required=""
            type="password"
            value=""
          />
          <label
            class="sc-htpNat gLmdTI"
            value=""
          >
            Confirm Password
          </label>
        </div>
        <button
          class="sc-bxivhb jzXiqt"
          type="submit"
        >
           SIGN UP 
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
